lmdeploy/serve/vl_async_engine.py

```diff
@@ -1,148 +1,197 @@
-                    features = [x['embeddings'] for x in features]
-
-                if isinstance(self.vl_prompt_template, MllamaTempateWrapper):
-                    # llama3.2 just encode <|image|> and inference
-                    decorated = decorated.replace(IMAGE_TOKEN, '<|image|>')
-                    input_ids = self.tokenizer.encode(decorated,
-                                                      add_bos=sequence_start)
-                    results['input_ids'] = input_ids
-                    results['prompt'] = decorated
-                    assert len(features)
-                    results['cross_attention_states'] = features[0]
-                    return results
-
-                if isinstance(self.vl_prompt_template,
-                              MolmoChatTemplateWrapper):
-                    return features[0]
-
-            features = [x.cpu().numpy() for x in features]
-            input_ids = []
-            begins = []
-            ends = []
-            if len(segs) != len(features) + 1:
-                logger.error(
-                    f'the number of {IMAGE_TOKEN} is not equal '
-                    f'to input images, {len(segs) - 1} vs {len(features)}')
-                features = features[:len(segs) - 1]
-            for i, seg in enumerate(segs):
-                if i > 0 and i <= len(features):
-                    image_dim = features[i - 1].shape[0]
-                    begins.append(len(input_ids))
-                    ends.append(begins[-1] + image_dim)
-                    input_ids.extend([IMAGE_DUMMY_TOKEN_INDEX] * image_dim)
-                seg_ids = self.tokenizer.encode(seg,
-                                                add_bos=((i == 0)
-                                                         and sequence_start))
-                input_ids.extend(seg_ids)
-            ranges = np.stack([begins, ends], axis=1).tolist()
-            results['input_embeddings'] = features or None
-            results['input_embedding_ranges'] = ranges or None
+        """process messages and return the required data for the inference
+        engines.
+
+        Refer to pytorch.engine.EngineInstance.async_stream_infer and
+        turbomind.TurboMindInstance.async_stream_infer for the argument
+        specification.
+        """
+        if isinstance(messages, str):
+            return super(self)._get_prompt_input(messages, do_preprocess,
+                                                 sequence_start, adapter_name,
+                                                 tools, **kwargs)
+        elif isinstance(messages, List):
+            has_multimodal_input = any(
+                isinstance(message['content'], list) and any(
+                    item['type'] in ['image_url', 'image_data']
+                    for item in message['content']) for message in messages)
+            if not has_multimodal_input:
+                return super(self)._get_prompt_input(messages, do_preprocess,
```

Comment:
failed when there's no input model data. 
1. should add type to super method, like `super(VLAsyncEngine, self)._get_prompt_input`
2. `_get_prompt_input` is coroutine and should be called with `await`, eg. `return await super(VLAsyncEngine, self)._get_prompt_input`
```
lmdeploy/lmdeploy/serve/vl_async_engine.py", line 75, in _get_prompt_input
    return super(self)._get_prompt_input(messages, do_preprocess,
TypeError: super() argument 1 must be type, not VLAsyncEngine
```